js/particle-modeler/index.js

```diff
@@ -129,6 +154,18 @@ export default class Interactive extends PureComponent {
       }
     });
 
+    api.onPropertyChange('time', function (t) {
+      // this will fire every tick
+      for (var i = 0, a; i < api.getNumberOfAtoms(); i++) {
+        a = api.getAtomProperties(i);
+        if (Math.abs(a.vx) > particleMaxVelocity || Math.abs(a.vy) > particleMaxVelocity) {
```

Comment:
This is allowing some particles to go faster than others so long as they travel diagonally. I assume it doesn't make much difference to the model, but you could always do vx-squared+vy-squared (and not bother with the sqrt, to save time).